extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -54,32 +59,58 @@ public void init(ByteBuffer buf, int position)
     createNewUnion(buf, position, false);
   }
 
+  /**
+   * This method uses locks because it can be used during indexing,
+   * and Druid can call aggregate() and get() concurrently
+   * https://github.com/apache/incubator-druid/pull/3956
+   */
   @Override
   public void aggregate(ByteBuffer buf, int position)
   {
     Object update = selector.getObject();
     if (update == null) {
       return;
     }
-
-    Union union = getOrCreateUnion(buf, position);
-    SketchAggregator.updateUnion(union, update);
+    final Lock lock = stripedLock.getAt(StripedLockHelper.lockIndex(position)).writeLock();
+    lock.lock();
+    try {
+      Union union = getOrCreateUnion(buf, position);
+      SketchAggregator.updateUnion(union, update);
+    }
+    finally {
+      lock.unlock();
+    }
   }
 
+  /**
+   * This method uses locks because it can be used during indexing,
+   * and Druid can call aggregate() and get() concurrently
+   * https://github.com/apache/incubator-druid/pull/3956
+   * The returned sketch is a separate instance of Sketch
+   * representing the current state of the aggregation, and is not affected by consequent
+   * aggregate() calls
```

Comment:
Please make `{@link #aggregate}` a Javadoc link

Reply:
Please make corresponding Javadoc fixes in all other parallel Aggregator classes